### PR TITLE
New running mode

### PR DESCRIPTION
**Pull request by NikolasHadjipanayi**

### CODE REVIEW

File: config/passport.js [hunk 1]
```diff
@@ -3,37 +3,49 @@ const bcrypt        = require('bcrypt-nodejs');
...
-					});
-
-				} else {
-					return done(null, false, {"message": "Invalid username."});
+    Users.count({},function(err, count) {
+		passport.use('login', new LocalStrategy(
+			function (usern, password, done) {
+				console.log(count);
+				if(count === 0)
+				{
```
Comment (Daanra) [inline config/passport.js]: This bracket is not placed in the usual spot

File: config/passport.js [hunk 2]
```diff
@@ -3,37 +3,49 @@ const bcrypt        = require('bcrypt-nodejs');
...
+        if(id === "firstlogin") {
+            var user = {_id:"firstlogin",username:"admin",password:"",rank:"admin"};
+            done(null, user);
+        }
+        else {
+            Users.findOne({_id: id}, function (err, user) {
+                if (typeof user === 'undefined' || err)
+                    console.log('Error in deserializing user');
+                done(null, user);
+            });
```
Comment (Daanra) [inline config/passport.js]: This file is now semi indented with tabs, semi with spaces

**Comment by mircealungu** [comment]
@Daanra  @NikolasHadjipanayi - are you guys waiting for me to merge the changes in the development?

**Comment by NikolasHadjipanayi** [comment]
@mircealungu No. I will fix the indentation mistakes that @Daanra mentioned and merge. Sorry for the delayed merges and approvals I was away for the break.

**Pull request merged**